### PR TITLE
[RLlib] fix marwil tf policy.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

Fix marwil TF policies. Also bring a test online.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/offline/json_reader.py [hunk 1]
```diff
@@ -47,11 +47,17 @@ def _adjust_obs_actions_for_policy(json_data: dict, policy: Policy) -> dict:
     Providing nested lists w/o this preprocessing step would
     confuse a SampleBatch constructor.
     """
-    for k, v in policy.view_requirements.items():
-        if k not in json_data:
-            continue
+    for k, v in json_data.items():
```
Comment (sven1977) [inline rllib/offline/json_reader.py:50]: Are these from another issue or related to MARWIL?

File: rllib/tests/test_nested_action_spaces.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def test_nested_action_spaces(self):
                 config["env_config"] = {
                     "action_space": action_space,
                 }
-                for flatten in [False, True]:
+                for flatten in [True, False]:
```
Comment (sven1977) [inline rllib/tests/test_nested_action_spaces.py:82]: Same question here. Does this fix the MARWIL issue or related to the failing nested_action_spaces test?

Comment (gjoliver) [reply rllib/tests/test_nested_action_spaces.py:82]: sorry this is no-op, I switched the 2 during testing. I can put them back though if you prefer.
and year, this is only for the failing nested_actions_space test.

Comment (sven1977) [reply rllib/tests/test_nested_action_spaces.py:82]: no, no worries, just wanted to understand it has something to do with the fix :)

**Review by sven1977 — APPROVED** [review]
Looks good to me. Thanks for the fix @gjoliver !

**Pull request merged**